vss-accessor/src/vss_error.rs

```diff
@@ -0,0 +1,75 @@
+use std::error::Error;
+use std::fmt::{Display, Formatter};
+use prost::{DecodeError, Message};
+use prost::bytes::Bytes;
+use reqwest::StatusCode;
+use crate::vss::{ErrorCode, ErrorResponse};
+
+#[derive(Debug)]
+pub enum VssError {
+    InvalidRequestError(ErrorResponse),
+    ConflictError(ErrorResponse),
+    InternalServerError(ErrorResponse),
+    InternalError(String),
+}
+
+impl VssError {
+    pub fn new(status: StatusCode, payload: Bytes) -> VssError {
+        match ErrorResponse::decode(&payload[..]) {
+            Ok(error_response) => VssError::from(error_response),
+            Err(e) => {
+                let message = format!(
+                    "Unable to decode ErrorResponse from server, HttpStatusCode: {}, DecodeErr: {}",
+                    status, e);
+                VssError::InternalError(message)
+            }
+        }
+    }
+}
+
+impl Display for VssError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match *self {
```

Comment:
No need to dereference here. Then you can remove the `ref` keywords below.

Reply:
fixed it